src/Compilers/CSharp/Portable/Symbols/Source/SourceFieldLikeEventSymbol.cs

```diff
@@ -33,6 +33,7 @@ internal SourceFieldLikeEventSymbol(SourceMemberContainerTypeSymbol containingTy
             _name = declaratorSyntax.Identifier.ValueText;
 
             var declaratorDiagnostics = DiagnosticBag.GetInstance();
+            Debug.Assert(declaratorSyntax.Parent is object);
```

Comment:
Consider moving this `Debug.Assert` to the top of the method body as it's effectively a pre-condition for the method itself, not this specific point in the method.